dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/deprecated/tracker/event/persistence/DefaultEventPersistenceService.java

```diff
@@ -112,6 +121,35 @@ public void delete(
     }
   }
 
+  @Override
+  @Transactional
+  public void updateEventDataValues(
+      EventDataValue de, org.hisp.dhis.dxf2.deprecated.tracker.event.Event event)
+      throws JsonProcessingException {
+
+    String uid = de.getDataElement();
+    de.setDataElement(null); // de uid is used as a key in the json, so we don't need it here
+
+    String query =
+        format(
+            "UPDATE event SET "
+                + "eventdatavalues= (CASE"
+                + "        WHEN eventdatavalues->:de IS NOT NULL"
+                + "        THEN jsonb_set(eventdatavalues, '{%1$s}', (eventdatavalues->:de) || '%2$s')"
+                + "        WHEN eventdatavalues->:de IS NULL"
+                + "        THEN jsonb_insert(eventdatavalues, '{%1$s}', '%2$s')"
+                + "    END) ,"
+                + " lastupdated = current_timestamp "
+                + "WHERE  uid = :event",
+            uid, mapper.writeValueAsString(de));
```

Comment:
Using a text block (tripple quotes) should make this more readable ;)

Reply:
ok thanks @jbee i will